src/api/listing/taxonomy_service.proto

```diff
@@ -0,0 +1,263 @@
+syntax = "proto3";
+package listing_api;
+
+import "api/listing/taxonomy.proto";
+import "google/api/annotations.proto";
+import "protoc-gen-openapiv2/options/annotations.proto";
+
+option go_package = "github.com/hdtradeservices/api-proto/go/api/listing";
+option php_generic_services = true;
+
+// TaxonomyService provides a service for managing the Taxonomy data
+// for a Listing integration.
+service TaxonomyService {
+  option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_tag) = {
+    name: "Listing Taxonomy Service",
+    // protolint:disable:next MAX_LINE_LENGTH
+    description: "Listing Taxonomy Service provides a service for managing the taxonomy data for a listing integration."
+  };
+
+  // ListCategories will return a list of all categories.
+  // If a parent_category_id is provided, only categories under that parent
+  // will be returned
+  rpc ListCategories(ListCategoriesRequest) returns (ListCategoriesResponse) {
+    option (google.api.http) = {
+      get: "/v2/storefront/taxonomy/category"
+    };
+  }
+
+  // CreateCategory will create a new Category
+  rpc CreateCategory(CreateCategoryRequest) returns (CreateCategoryResponse) {
+    option (google.api.http) = {
+      post: "/v2/storefront/taxonomy/category",
+      body: "*"
+    };
+  }
+
+  // UpdateCategory will update the Category with the given ID
+  rpc UpdateCategory(UpdateCategoryRequest) returns (UpdateCategoryResponse) {
+    option (google.api.http) = {
+      put: "/v2/storefront/taxonomy/category",
+      body: "*"
+    };
+  }
+
```

Comment:
there's no `DeleteCategory`

Reply:
hm we dont have one on the internal service yet either, maybe we just build it once its required.